filebeat/crawler/prospector_log.go

```diff
@@ -48,32 +46,37 @@ func (p *ProspectorLog) Init() {
 
 		// Offset found -> skip to previous state
 		if offset > 0 {
-			fs := input.NewFileStat(fileinfo, 0)
+			fs := input.NewFileState(fileinfo)
 			fs.Skip(offset)
-			p.harvesterStats[path] = *fs
+			p.harvesterStates[path] = *fs
```

Comment:
with passing FileState by value, how about changing NewFileState to return a value instead of a pointer? Maybe rename to 'MakeFileState'?